engine/collection/compliance/core.go

```diff
@@ -301,7 +301,7 @@ func (c *Core) processBlockProposal(proposal *messages.ClusterBlockProposal) err
 
 	// submit the model to hotstuff for processing
 	log.Info().Msg("forwarding block proposal to hotstuff")
-	c.hotstuff.SubmitProposal(header, parent.View)
+	<-c.hotstuff.SubmitProposal(header, parent.View)
```

Comment:
I think I should remove this. Or we could keep it and always wait to process blocks in the compliance engine.

Reply:
Yeah, let's replace it with a TODO:
```suggestion
	// TODO: wait for the returned callback channel if we are processing blocks from range response
	c.hotstuff.SubmitProposal(header, parent.View)
```